frontend/packages/kubevirt-plugin/src/components/modals/menu-actions-modals/utils.ts

```diff
@@ -1,12 +1,20 @@
 import { history } from '@console/internal/components/utils';
-import { VIRTUALMACHINES_BASE_URL } from '../../../constants/url-params';
+import {
+  VIRTUALMACHINES_BASE_URL,
+  VIRTUALMACHINES_TEMPLATES_BASE_URL,
+} from '../../../constants/url-params';
 import { getName, getNamespace } from '../../../selectors';
 import { VMGenericLikeEntityKind } from '../../../types/vmLike';
 
-export const redirectToList = (vmi: VMGenericLikeEntityKind, tab?: 'templates' | '' | null) => {
+export const redirectToList = (vmi: VMGenericLikeEntityKind, isTemplate?: boolean) => {
```

Comment:
notes:
In https://github.com/openshift/console/pull/10338 we changed the template list url and broke this method.

When revisiting it note this is actually a "redirect to list after delete" if invoked from the deleted resource page.
you can look at a reference implementation here in the generic delete modal:
https://github.com/openshift/console/blob/bec15d9d322a8f8eb40fcefbca52bdfa8303b0de/frontend/public/components/modals/delete-modal.tsx#L40

They use `resourcePathFromModel` to calculate the list URL:
https://github.com/openshift/console/blob/bec15d9d322a8f8eb40fcefbca52bdfa8303b0de/frontend/public/components/utils/resource-link.tsx#L22

In this case, because templates do not follow the OCP URL pattern you will need to use `(crd=false, plural='virtualmachinetemplates')` to use the generic method, or create a custom one.

a custom method can look like:
```
const resourceListPath = (namespace: string, resourceRef: string) => `/k8s/ns/${namespace}/${resourceRef}`
```
Note that:
`resourceRef` for VMs is the usual `kubevirtReferenceForModel(VirtualMachineSnapshotModel)`
while `resourceRef` is `virtualmachinetemplates`

IMHO a nice future refactor will be to change the method name and parameters to better explain the method use case, and use internal parameters to beter clear how it's done:
``` js
const redirectToListAfterDelete = (obj: VMGenericLikeEntityKind) => {
...
if (...) {
const resourceRef = obj?.kind === TemplateModel.kind ? 'virtualmachinetemplates' : referenceForModel(model);
const namespace = obj?.metadata?.namespace;|
history.push(`/k8s/ns/${namespace}/${resourceRef}`)
}
...
```